Makefile

```diff
@@ -141,22 +142,27 @@ endif
 
 # build the chain binary in docker
 build-chain:
-	docker run -i --rm \
-		-v ${HOME}/.cargo/git:/root/.cargo/git \
-		-v ${HOME}/.cargo/registry:/root/.cargo/registry \
-		-v `pwd`:/chain \
-		--env RUSTFLAGS=-Ctarget-feature=+aes,+sse2,+sse4.1,+ssse3 \
-		--workdir=/chain \
-		$(IMAGE_RUST):latest \
-		bash -c '. /root/.docker_bashrc && \
-		echo "========  build dev-utils   =========" && \
-		${CARGO_BUILD_CMD} --bin dev-utils && \
-		echo "========  build chain-abci   =========" && \
-		cd chain-abci && $(CARGO_BUILD_CMD_ABCI) && \
-		echo "========  build client-cli   =========" && \
-		cd ../client-cli && $(CARGO_BUILD_CMD_CLI)&& \
-		echo "========  build client-rpc   =========" && \
-		cd ../client-rpc && $(CARGO_BUILD_CMD_CLI)'
+	@if [ -e "./target/${BUILD_MODE}/client-cli" ] && [ -e "./target/${BUILD_MODE}/chain-abci" ] && [ -e "./target/${BUILD_MODE}/client-rpc" ] && [ -e "./target/${BUILD_MODE}/dev-utils" ]; then \
```

Comment:
I think it's quite dangerous to use the same target directory for `make build` and `cargo build`. If the developer uses them both on different platforms, the resulting binary might be undefined?
But this issue is out of the scope of this PR.

Reply:
true; this may not be reproducible -- so even though one person follows the same instructions on one platform, it may not work.

Reply:
target folder is kindof mutex, simultaneous compiling is not performed.